components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -646,6 +646,17 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
         });
     }
 
+    public async countUnabortedPrebuildsSince(cloneURL: string, date: Date): Promise<number> {
+        const abortedState: PrebuiltWorkspaceState = 'aborted';
+        const repo = await this.getPrebuiltWorkspaceRepo();
+
+        let query = repo.createQueryBuilder('pws');
+        query = query.where('pws.cloneURL = :cloneURL', { cloneURL })
```

Comment:
Checking [DBPrebuiltWorkspace](https://github.com/gitpod-io/gitpod/blob/6081ee1dad74f863a45f4e3fc329a830e59ec5a5/components/gitpod-db/src/typeorm/entity/db-prebuilt-workspace.ts#L14) I noticed we don't have an index for `(cloneURL, creationTime, state)`. We should definitely add one before merging. Here's an example of how to add a migration for a secondary index: https://github.com/gitpod-io/gitpod/commit/1c4cd73f47e6c92941a0516ea3f746d5a7ef900b

Reply:
Great spot! Will go through this shortly.

Reply:
Added the migrations.